### PR TITLE
Don't allow locked users to roompromote

### PR DESCRIPTION
**Pull request by panpawn**

### CODE REVIEW

File: chat-commands.js [hunk 1]
```diff
@@ -1052,6 +1052,7 @@ exports.commands = {
 			this.sendReply("/roompromote - This room isn't designed for per-room moderation");
 			return this.sendReply("Before setting room mods, you need to set it up with /roomowner");
 		}
+		if (!this.canTalk() && !user.can('bypassall')) return;
```
Comment (Zarel) [inline chat-commands.js]: You don't need to do `bypassall` separately; I think there's a `bypassall` check in `canTalk` (and if there isn't, it should be added there)

**Comment by panpawn** [comment]
Updated

**Comment by panpawn** [comment]
.... I can't tell if I'm doing something wrong or if I should change something in the tests of what...

**Comment by Zarel** [comment]
Try running `npm test` on master?

**Comment by Zarel** [comment]
no, it definitely is in this specifically

**Comment by Zarel** [comment]
Apparently admins can't actually bypass `canTalk`?

**Comment by panpawn** [comment]
https://github.com/Zarel/Pokemon-Showdown/blob/master/chat.js#L533

??

**Comment by Zarel** [comment]
Yeah, clearly one step of that is failing. Run it through a debugger (or just do print-debugging).

**Comment by panpawn** [comment]
Hm...

When I had this as ``if (!this.canTalk() && !user.can('bypassall')) return;`` it passed Travis testing. 

I can't figure out the issue in ``canTalk``, so should I just revert this back to that line?

**Comment by Zarel** [comment]
No, like I said, run it through a debugger (or just do print-debugging).

If you can't debug it, idk, wait for me to have free time to debug it or something.

**Comment by panpawn** [comment]
Someone else can handle this, I'm on a LoA right now.

And this should probably be handled sooner rather than later.

**Pull request closed without merging**